cdap-common/src/main/java/co/cask/cdap/common/conf/Constants.java

```diff
@@ -266,6 +268,8 @@
       /** for the address (hostname) of the dataset server. */
       public static final String ADDRESS = "dataset.service.bind.address";
       public static final String PORT = "dataset.service.bind.port";
+      public static final String ANNOUNCE_ADDRESS = "dataset.service.announce.address";
```

Comment:
i am not sure, depends on the use case. it doesn't make sense to have dataset.bind.address and app.bind.address, we should consolidate that, for now i have updated to use single announce address.